### PR TITLE
feat(rust): use cli state for trust context and default trust context

### PR DESCRIPTION
**Pull request by caoakleyii**
## Current behavior

- Trust Context relies on a JSON file always
- No Default Trust Context
- Trust Context parameter can only be a json file

## Proposed changes
- `ockam trust-context create` 
    -  Creates and stores a trust context from the Project Files/States, Credentials, Trust Context JSON files, and auto defaults to the default project
- removes `--as-trust-context` option from project and credential
    - A user can use the following to create and store a trust context
    - `ockam trust-context create [name]` (defaults to default project)
    - `ockam trust-context create [name] --project [PROJECT_NAME]`
    - `ockam trust-context create [name] --credential [CREDENTIAL_NAME]`
- Trust Context is stored within the CLI state
- Trust Context has a default option
- Commands using `--trust-context` [ockam node create, enroll, auth, etc] will use the default trust context if available
- `--trust-context` option accepts both a path and cli state name

 

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

**Comment by caoakleyii** [comment]
A more verbose and explicit example usages would look like:

```sh
ockam enroll

# create and store a trust context named `my-context` using the default project
ockam trust-context create my-context
ockam node create --trust-context my-context
```

However, with the default options; this can be boiled down to:


```sh
ockam enroll # enroll creates a default project, and trust context
ockam node create # node create knows to use the default project & trust context
```

File: implementations/rust/ockam/ockam_api/src/cli_state.rs [hunk 1]
```diff
@@ -1238,6 +1245,97 @@ impl ProjectState {
...
+        let link = self.default_path()?;
+        std::os::unix::fs::symlink(original, link)?;
+        self.get(name)
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct TrustContextState {
+    pub path: PathBuf,
+}
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_api/src/cli_state.rs]: Is there any particular reason not to include the config here? I saw the same pattern in the `ProjectState` which differs from the rest of the states. I've found it useful to be able to have the decoded config as part of the state, but maybe there is a reason not to do that?

Comment (caoakleyii) [reply implementations/rust/ockam/ockam_api/src/cli_state.rs]: I will update this.

Regarding ProjectState I believe there was a reason at one point, however I don't believe it's applicable anymore.

Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_api/src/cli_state.rs]: If it's not necessary for this PR we can add that on a separate PR. Up to you!

Comment (caoakleyii) [reply implementations/rust/ockam/ockam_api/src/cli_state.rs]: No big deal to add for TC. I want to refactor this anyways, so i'm going to leave project state for now.

**Review by adrianbenavides — APPROVED** [review]

**Pull request merged**